compiler/rustc_type_ir/src/sty.rs

```diff
@@ -18,6 +18,27 @@ use self::TyKind::*;
 use rustc_data_structures::stable_hasher::HashStable;
 use rustc_serialize::{Decodable, Decoder, Encodable};
 
+/// Specifies how a trait object is represented.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Encodable, Decodable)]
+pub enum DynKind {
+    /// An unsized `dyn Trait` object
+    Dyn,
+    /// A sized `dyn* Trait` object
```

Comment:
It would be good to say a bit about how such an object is represented.

Reply:
Done. Let me know if the explanation makes sense.